noun/retrieve.c

```diff
@@ -1770,113 +1518,122 @@ u3_mur_words(const c3_w* key_w, c3_w len_w)
   }
 }
 
-/* u3_mur_both():
+/* u3r_mug_bytes(): Compute the mug of `buf`, `len`, LSW first.
 **
-**   Join two murs.
+**   XX temporary, reconcile with u3r_mug_words
+*/
+c3_w
+u3r_mug_bytes(const c3_y *buf_y,
+              c3_w        len_w)
+{
+  c3_w syd_w = 0xcafebabe;
+
+  while ( 1 ) {
+    c3_w haz_w, ham_w;
+
+    MurmurHash3_x86_32(buf_y, len_w, syd_w, &haz_w);
+    ham_w = (haz_w >> 31) ^ (haz_w & 0x7fffffff);
+
+    if ( 0 != ham_w ) return ham_w;
+    else syd_w++;
+  }
+}
+
+/* u3r_mug_d(): Compute the mug of `num`, LSW first.
+*/
+c3_w
+u3r_mug_d(c3_d num_d)
+{
+  c3_w buf_w[2];
+
+  buf_w[0] = (c3_w)(num_d & 0xffffffffULL);
+  buf_w[1] = (c3_w)(num_d >> 32ULL);
+
+  return u3r_mug_words(buf_w, 2);
+}
+
+/* u3r_mug_both(): Join two mugs.
 */
 c3_w
-u3_mur_both(c3_w lef_w, c3_w rit_w)
+u3r_mug_both(c3_w lef_w, c3_w rit_w)
 {
   c3_w ham_w = lef_w ^ (0x7fffffff ^ rit_w);
 
-  return u3_mur_words(&ham_w, (0 == ham_w) ? 0 : 1);
+  return u3r_mug_words(&ham_w, (0 == ham_w) ? 0 : 1);
 }
 
-/* u3_mur(): MurmurHash3 on a noun.
+/* u3r_mug(): memoized 31-bit nonzero MurmurHash3 on a noun.
 */
 c3_w
-u3_mur(u3_noun veb)
+u3r_mug(u3_noun veb)
 {
-  if ( u3_fly_is_cat(veb) ) {
-    return u3_mur_words(&veb, (0 == veb) ? 0 : 1);
+  c3_assert(u3_none != veb);
+
+  if ( _(u3a_is_cat(veb)) ) {
+    return u3r_mug_words(&veb, (0 == veb) ? 0 : 1);
```

Comment:
This looks like the same pattern as in u3r_mug_both(). Seems like the pattern should be moved into u3r_mug_words()